source/clear-linux/guides/maintenance/autospec.rst

```diff
@@ -0,0 +1,316 @@
+.. _autospec:
+
+Build RPMs with autospec
+########################
+
+This guide shows you how to create RPMs with :ref:`autospec <autospec-about>`
+, a tool that assists in automated creation and maintenance of RPM packaging 
+on |CLOSIA|. Additionally, you will use the RPMs to create bundles using 
+:ref:`mixer <mixer>`. 
+
+Prerequisites
+*************
+This guide assumes that you have:
+
+* Created :ref:`a custom mix <mixer>` of |CL| and deployed it to a
+  to a target device
+
+* |CL| running on a host machine or virtual environment
+  
+  .. note:: 
+
+     To install |CL|, see: 
+
+     * :ref:`bare-metal-install`
+     * :ref:`virtual-machine-install`
+   
+Install Clear Linux tooling framework
+=====================================
+
+Our GitHub\* repository provides you with the resources you need
+to create and maintain packages.
+
+#. On your host system, install this developer bundle. 
+   
+   .. code-block:: bash
+
+      sudo swupd bundle-add os-clr-on-clr
+
+#. Run this command to download the :file:`user-setup.sh` script.
+
+   .. code-block:: bash
+
+      curl -O https://raw.githubusercontent.com/clearlinux/common/master/user-setup.sh
+   
+#. Make :file:`user-setup.sh` executable.
+
+   .. code-block:: bash
+
+      chmod +x user-setup.sh
+
+#. Run the script as an unprivileged user.
+  
+   .. code-block:: bash
+
+      ./user-setup.sh
+
+#. After the script completes, log out and log in again to complete
+   the setup process.
+
+   The `user-setup script`_ creates a folder called :file:`clearlinux`, which
+   contains the :file:`Makefile`, :file:`packages`, and :file:`projects`
+   subfolders.  
+
+   The :file:`projects` folder contains the main tools, `autospec` 
+   and `common`, used for making packages in |CL|. 
+
+Create a RPM with autospec
+**************************
+
+.. include:: ../../concepts/autospec-about.rst
+   :Start-after: autospec-overview
+   :end-before: For a comprehensive list of control files
+
+For a detailed explanation of how ``autospec`` works on |CL|, visit our 
+:ref:`autospec-about`.  For a general understanding of how RPMs work, we 
+recommend visiting the `rpm website`_ or the `RPM Packaging Guide`_ . 
+
+Building RPMs 
+=============
+
+Choose one of the following options to build RPMs and manage source
+code: 
+
+* :ref:`build-a-new-rpm` and spec file using ``make autospecnew``. 
+
+* :ref:`build-source-code-with-existing-spec-file` (without changing the
+  spec file) using ``make build``.  
+
+* :ref:`generate-a-new-spec-file` based on changes in the control files with
+  ``make autospec``. 
+
+.. _build-a-new-rpm:
+
+Build a new RPM
+===============
+
+#. Navigate to the autospec workspace.
+
+   .. code-block:: bash
+
+      cd ~/clearlinux
+
+#. Enter the command: 
+
+   .. code-block:: bash
+
+      make autospecnew URL="https://github.com/bktan/helloclear/archive/helloclear-4.tar.gz" NAME="helloclear"
```

Comment:
URL should be `https://github.com/bktan8/helloclear/archive/helloclear-4.tar.gz`